src/curl.c

```diff
@@ -93,16 +101,24 @@ int swupd_curl_init(void)
 				continue;
 			}
 			fallback_capaths[i] = strdup(tok);
-			if (++i > CA_PATH_NO_MAX) {
-				fprintf(stderr, "Warning: up to %d CApaths can be specified."
-						" Ignoring the rest.\n", CA_PATH_NO_MAX);
-				break;
+			if (++i > sz) {
+				char **re_fallback_capaths, **t;
+				sz <<= 1;
+				re_fallback_capaths = (char **)malloc(sz*sizeof(char *));
+				memcpy(re_fallback_capaths, fallback_capaths, i*sizeof(char *));
```

Comment:
Also this is almost exactly the behavior of realloc (malloc new array, copy values, free old one). Now it is true that realloc has some issues if you want to fail gracefully if you run out of memory, but swupd client is not one of those programs (see uses of string_or_die everywhere). I would just use the library function.